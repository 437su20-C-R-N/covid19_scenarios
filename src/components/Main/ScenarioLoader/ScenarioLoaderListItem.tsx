import React from 'react'

import { NavLink } from 'reactstrap'

import './ScenarioLoader.scss'

export interface ScenarioOption {
  label: string
  value: string
}

export interface ScenarioLoaderListItemProps {
  option: ScenarioOption
  onItemClick: (id: string) => void
}

export function ScenarioLoaderListItem({ option, onItemClick }: ScenarioLoaderListItemProps) {
  return (
    <div className="scenario-loader-row">
      <NavLink
        data-testid="PresetLoaderDialogRowLink"
        onClick={(event) => {
          event.preventDefault()
          onItemClick(option.value)
        }}
      >
        {option.label}
      </NavLink>
    </div>
  )
}
