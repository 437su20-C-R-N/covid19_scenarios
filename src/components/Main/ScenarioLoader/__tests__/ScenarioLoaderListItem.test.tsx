import React from 'react'

import { noop } from 'lodash'
import { cleanup, fireEvent, render, RenderResult } from '@testing-library/react'

import { ScenarioLoaderListItem, ScenarioOption } from '../ScenarioLoaderListItem'

const DATA: ScenarioOption = {
  label: 'Some label',
  value: '123',
}

describe('ScenarioLoaderListItem', () => {
  let onLoadButtonClick: (id: string) => void
  let wrapper: RenderResult

  beforeEach(() => {
    console.warn = noop
    console.error = noop

    onLoadButtonClick = jest.fn()
    wrapper = render(<ScenarioLoaderListItem option={DATA} onItemClick={onLoadButtonClick} />)
  })

  afterEach(cleanup)

  it('matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders label and button', () => {
    expect(wrapper.getByText('Some label')).not.toBeNull()
    expect(wrapper.getByTestId('PresetLoaderDialogRowLink')).not.toBeNull()
    expect(wrapper.getByTestId('PresetLoaderDialogRowButton')).not.toBeNull()
  })

  it('link click triggers onLoadButtonClick handler', () => {
    const loadButton = wrapper.getByTestId('PresetLoaderDialogRowLink')
    fireEvent.click(loadButton)

    expect(onLoadButtonClick).toHaveBeenCalledWith('123')
  })

  it('button click triggers onLoadButtonClick handler', () => {
    const loadButton = wrapper.getByTestId('PresetLoaderDialogRowButton')
    fireEvent.click(loadButton)

    expect(onLoadButtonClick).toHaveBeenCalledWith('123')
  })
})
