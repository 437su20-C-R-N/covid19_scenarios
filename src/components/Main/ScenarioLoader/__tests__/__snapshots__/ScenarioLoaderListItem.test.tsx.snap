// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScenarioLoaderListItem matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="scenario-loader-row"
  >
    <a
      class="nav-link"
      data-testid="PresetLoaderDialogRowLink"
    >
      Some label
    </a>
  </div>
</DocumentFragment>
`;
